### PR TITLE
Initialize data providers in Benchmark

### PR DESCRIPTION
**Pull request by orestisfl**
### Summary of your changes

  - Refactor the data providers so that they are initialized by each Benchmark
  - Remove switches on cfg.Benchmark
  - Merge awslib.Identity with account alias
      - Remove need to initialize IAM fetcher to get AWS alias

### Related Issues
- Related: https://github.com/elastic/cloudbeat/pull/1093
- Related: https://github.com/elastic/cloudbeat/issues/1039

### Checklist
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added the necessary README/documentation (if appropriate)

### CODE REVIEW

File: dataprovider/providers/aws/option.go [hunk 1]
```diff
@@ -29,9 +31,9 @@ func WithLogger(log *logp.Logger) Option {
 	}
 }
 
-func WithAccount(name string, id string) Option {
+func WithAccount(identity *awslib.Identity) Option {
```
Comment (olegsu) [inline dataprovider/providers/aws/option.go:34]: remove pointer reference

**Review by uri-weisman — CHANGES_REQUESTED** [review]
Love it

File: dataprovider/providers/aws/option.go [hunk 2]
```diff
@@ -29,9 +31,9 @@ func WithLogger(log *logp.Logger) Option {
...
 	return func(dp *DataProvider) {
-		dp.accountId = id
-		dp.accountName = name
+		dp.accountId = identity.Account
+		dp.accountName = identity.Alias
```
[the remark this answers is not in the document]
Comment (orestisfl) [reply dataprovider/providers/aws/option.go:37]: Yes :+1:

File: resources/providers/awslib/identity_provider.go [hunk 3]
```diff
@@ -35,17 +38,32 @@ type IdentityProviderGetter interface {
 type IdentityProvider struct{}
 
 // GetIdentity returns AWS identity information
-func (provider IdentityProvider) GetIdentity(ctx context.Context, cfg aws.Config) (*Identity, error) {
-	client := sts.NewFromConfig(cfg)
+func (p IdentityProvider) GetIdentity(ctx context.Context, cfg aws.Config) (*Identity, error) {
```
[the remark this answers is not in the document]
Comment (orestisfl) [reply resources/providers/awslib/identity_provider.go:41]: Reason is for them to be mockable

**Review by uri-weisman — APPROVED** [review]

**Pull request merged**